### PR TITLE
Modified view-agency component & css for updated interface

### LINKED ISSUE #212: Officers in the Agency Admin Interface
When an agency administrator clicks Agencies in the top nav - and then clicks on an agency - they get to the /agencies/view_agency page. The first tab is 'Officers' and currently shows 0 officers: 
Screen Shot 2020-10-02 at 12 53 46 PM

As per the design at https://mongodb.invisionapp.com/share/MYXRXC2T2H3#/screens/424650597 the page should show all the users in the agency. 

There should be a count of officers; a search bar; and an 'Add New User' button. Then a list of officers should display sorted alphabetically - ascending by last name. Each officer's row should display: Name - Group - Role - Status and a link to 'Edit' the user. For this issue to be considered complete the search bar does NOT have to work. The 'Add New User' link should work and go to the /users/new_user page.

### PR DESCRIPTION
**Pull request by deveshchatuphale7**
## Related Issue

Fixes #212 

## Checklist:

- [x] I have read the [contributor's guide](https://wildaid.github.io/contribute/index.html).
- [x] I linked an issue in the previous section
- [x] I have commented on the linked issue
- [x] I was assigned the linked issue (not required)
- [x] I have tested the change to the best of my ability against the [sandbox](https://wildaid.github.io/contribute/sandbox.html) or a [local build](https://wildaid.github.io/build).

Optional items:

- [ ] My change adds new text and requires a change to translations.
- [ ] My change requires a change to the documentation.
- [ ] I have submitted a PR to the [documentation repo](https://github.com/WildAid/wildaid.github.io).
- [ ] I was not able to test... (explain below, e.g. you did not have permissions to test a specific feature)
- [ ] This change depends O-FISH Realm repository changes (explain below)

* **Optional: Add any explanations here** 
Modified view-agency component as well as css file to incorporate design requirements 

* **Optional: Add any relevant screenshots here**

### CODE REVIEW

File: src/components/agencies/view-agency/view-agency.component.js [hunk 1]
```diff
@@ -60,39 +79,32 @@ class ViewAgency extends Component {
...
-                    />
-                  </div>
-                  {agency.email}
-                </div>
-              </Fragment>
-            )}
+                </Fragment>
+              )}
           </div>
           <NavLink to={EDIT_AGENCIES_PAGE.replace(":id", agency._id)}>
```
Comment (Sheeri) [inline src/components/agencies/view-agency/view-agency.component.js:109]: This line is giving me an error when I run npm start locally on this branch:

Failed to compile.

./src/components/agencies/view-agency/view-agency.component.js
  Line 109:58:  'agency' is not defined  no-undef

**Review by Sheeri — CHANGES_REQUESTED** [review]
I cannot get this to build....see inline comment.

**Comment by Sheeri** [comment]
Hi, it still won't build - you can see that on this PR page:
Screen Shot 2020-10-22 at 1 28 00 PM

and if you click through to "Details" you can see:

./src/components/agencies/view-agency/view-agency.component.js
  Line 257:41:  'agencyInfo' is not defined  no-undef

**Review by Sheeri — CHANGES_REQUESTED** [review]
Hi, it still won't build - you can see that on this PR page:
Screen Shot 2020-10-22 at 1 28 00 PM

and if you click through to "Details" you can see:

./src/components/agencies/view-agency/view-agency.component.js
  Line 257:41:  'agencyInfo' is not defined  no-undef

**Review by Sheeri — CHANGES_REQUESTED** [review]
Hi @deveshchatuphale7 - Can you see the "checks" section of https://github.com/WildAid/o-fish-web/pull/326 ? I can't review the code until it builds and gets a green checkmark there. It looks like you fixed the first issue, but now there's another issue.

**Comment by deveshchatuphale7** [comment]
> 
> 
> Hi @deveshchatuphale7 - Can you see the "checks" section of #326 ? I can't review the code until it builds and gets a green checkmark there. It looks like you fixed the first issue, but now there's another issue.

Thanks for informing. Hopefully my latest commit will fix the issue. But I am curious why I was not getting any issue while generating build locally

**Comment by Sheeri** [comment]
Hi there! The build on the website builds the *current code* with your changes. You probably did not do a "git pull" before your testing and PR submission, so you had older code - the files you didn't change. Github tests the current code plus your changes because there could be issues and it wants to find them first.

**Review by Sheeri — CHANGES_REQUESTED** [review]
Hi! It built, but I'm not seeing the problem resolved. 

I logged in as an Agency Administrator, then went to the Agencies page. Then clicked on my agency, and here's the page I got:


Screen Shot 2020-10-23 at 2 08 01 PM


It still shows 0 officers, when there should at least be 1 officer (me!).

I did this against the sandbox; is that what you used?

**Comment by deveshchatuphale7** [comment]
> 
> 
> Hi! It built, but I'm not seeing the problem resolved.
> 
> I logged in as an Agency Administrator, then went to the Agencies page. Then clicked on my agency, and here's the page I got:
> 
> Screen Shot 2020-10-23 at 2 08 01 PM
> 
> It still shows 0 officers, when there should at least be 1 officer (me!).
> 
> I did this against the sandbox; is that what you used?

Yes, Actually also for my login I got officers array with key officers under agencyAdditionalInfo object was empty. Hence dashboard is showing zero officers. 

image

Did I miss any step here, could you point me in right direction ?

**Comment by Sheeri** [comment]
Ah! I see. You correctly printed out agencyAdditionalInfo.

The problem is that agencyAdditionalInfo object is wrong - how is it populated? is it looking in the wrong place for officers? 

You might be able to find some clues by looking at src/components/agencies/agencies.component.js - I can see there's a function in there called getAgenciesWithOfficers, which is how that page gets a list of officers, which it then uses to display.

File: src/components/agencies/view-agency/view-agency.component.js [hunk 2]
```diff
@@ -30,16 +41,36 @@ class ViewAgency extends Component {
     });
   };
 
+  goEditUser = (id) => {
+    //TODO: Use router!
```
Comment (Sheeri) [inline src/components/agencies/view-agency/view-agency.component.js:45]: Can you say more about this "TODO"? what should be done?

File: src/components/agencies/view-agency/view-agency.component.js [hunk 3]
```diff
@@ -128,44 +160,112 @@ class ViewAgency extends Component {
...
+                      </div>
+                      <input
+                        className="search-field officer-search-panel-input"
+                        type="search"
+                        placeholder={`${t("SEARCH.FILTER_SEARCH")} O-FISH`}
+                        onChange={this.setSearch}
+                        onFocus={() => this.setState({ isFocused: true })}
+                      ></input>
+                    </div>
+
```
Comment (Sheeri) [inline src/components/agencies/view-agency/view-agency.component.js:182]: Is search supposed to work? You added the UI components, which is awesome. I'm happy to make another ticket for search to work, but if you intended it to work it does not. Just want to make sure we're on the same page!

File: src/components/agencies/view-agency/view-agency.component.js [hunk 4]
```diff
@@ -30,16 +41,36 @@ class ViewAgency extends Component {
...
     this.setState({ loading: true }, () => {
       agencyService
         .getAgency(id)
         .then((data) => {
           this.setState({
             agency: data,
-            loading: false,
+          });
+
+          boardingService.getBoardingsWithFacet(50,0,null,{"agency":data.name}).then((userData)=>{
```
Comment (Sheeri) [inline src/components/agencies/view-agency/view-agency.component.js:59]: If I'm reading this properly, this gets a list of officers by getting the boarding records for the agency, and finding all the distinct reportingOfficer email addresses. 

But we want to see officers, even if they have not submitted any boarding records.

This should use the "searchFacetByUsers" function, which takes in a limit, offset, query, and filter, and returns an array of users (using limit and offset to only return what's needed on the page), a total count of users, and any highlighting needed. 

You can see that function at https://github.com/WildAid/o-fish-realm/blob/main/WildAidDemo/functions/searchFacetByUsers/source.js

**Review by Sheeri — COMMENTED** [review]
Hi @deveshchatuphale7 - this is a great start! I'm getting a count and a list of users, but it's not getting them from the right place. I added some inline comments and suggestions. 

I'm also not sure if search is supposed to work? Right now it doesn't - if I type something into the search bar and hit "return" nothing happens. It's totally OK if search doesn't work, but it looks like the code intends it to be working, perhaps it is supposed to highlight a search term.

I'm happy to keep the scope of this to just the count and list of officers, and we can make a different ticket for search to work. (but then we should remove the code related to search functionality - though it's OK to keep the UI code in there - it looks great!)

File: src/components/agencies/view-agency/view-agency.component.js (hunk 3 above)
Comment (deveshchatuphale7) [reply src/components/agencies/view-agency/view-agency.component.js:182]: I think for current issue search functionality doesn't have to work & also separate ticket for search, issue #213 is already present, which is dependent on issue #212

File: src/components/agencies/view-agency/view-agency.component.js (hunk 2 above)
Comment (deveshchatuphale7) [reply src/components/agencies/view-agency/view-agency.component.js:45]: Sorry I forgot to remove the comment, I have added logic  `goToPage(EDIT_USER_PAGE, id);` under the function, which navigates to edit user page.

File: src/components/agencies/view-agency/view-agency.component.js (hunk 4 above)
Comment (deveshchatuphale7) [reply src/components/agencies/view-agency/view-agency.component.js:59]: I tried using getUsers from users.service which calls searchFacetByUsers function. With agency name as filter I received empty array in response & without any filters I received list of users, In my case I received user details of me & another user I created. So what parameters would be helpful to get officer details ?

Comment (Sheeri) [reply src/components/agencies/view-agency/view-agency.component.js:59]: Ah, you don't need a filter, there are permissions in place to make sure you only see what you're allowed to see. (as an agency admin, you can only see users in your agency)

You say you get the user details for you and the other user you created, which sounds like you have the right link there. What additional details do you need?

Comment (deveshchatuphale7) [reply src/components/agencies/view-agency/view-agency.component.js:59]: Yes I think response from getUsers will suffice the requirement. but one doubt I have is, on agencies page for agency name deveshchatuphale7 respective officer name in the list is Emex Geefy.But, I didn't receive any info about that user in the response of getUsers without any filters & searchQuery.
So just wanted to know if I am receiving a correct response or I have to pass any additional filters or searchQuery ?

Comment (Sheeri) [reply src/components/agencies/view-agency/view-agency.component.js:59]: To debug, loop through every part of the returned user object, and print it, to see where the issue is. Perhaps there is a different field you should be using? (that's what I'd do, I don't know where the problem is here).

**Review by Sheeri — APPROVED** [review]
This looks fantastic!

**Comment by Sheeri** [comment]
Congratulations! 🎉 🙌 🎆 ㊗️ 

This PR is great and has been merged into the code. Thanks so much for contributing! If you'd like, you can request to work on another issue. We really appreciate your effort!

To claim your O-FISH badge, head on over to the [MongoDB Community forums](https://developer.mongodb.com/community/forums/t/open-source-contributors-pick-up-your-badges-here/10167/9), login, and reply - making sure to link to this PR.

**Comment by deveshchatuphale7** [comment]
> 
> 
> Congratulations! 🎉 🙌 🎆 ㊗️
> 
> This PR is great and has been merged into the code. Thanks so much for contributing! If you'd like, you can request to work on another issue. We really appreciate your effort!
> 
> To claim your O-FISH badge, head on over to the [MongoDB Community forums](https://developer.mongodb.com/community/forums/t/open-source-contributors-pick-up-your-badges-here/10167/9), login, and reply - making sure to link to this PR.

Sure!

**Pull request merged**